haystack/components/evaluators/llm_evaluator.py

```diff
@@ -0,0 +1,230 @@
+import json
+from typing import Any, Dict, List, Optional, Tuple, Type
+
+from haystack import component, default_from_dict, default_to_dict
+from haystack.components.builders import PromptBuilder
+from haystack.components.generators import OpenAIGenerator
+from haystack.utils import Secret, deserialize_secrets_inplace
+
+
+@component
+class LLMEvaluator:
+    """
+    Uses an LLM to evaluate inputs based on a prompt containing instructions and examples.
+
+    The default API requires an OpenAI API key to be provided as an environment variable "OPENAI_API_KEY".
+    The inputs are lists that are user-defined depending on the desired metric.
+    The output is a dictionary with a key `results` containing a list of evaluation results.
+    Each result is a dictionary with user-defined keys and values of either 0 for FALSE or 1 for TRUE respectively.
+
+    Usage example:
+    ```python
+    from haystack.components.evaluators import LLMEvaluator
+    evaluator = LLMEvaluator(
+        instructions="Is this answer problematic for children?",
+        inputs=[("responses", List[str])],
+        outputs=["score"],
+    )
+    RESPONSES = [
+        "Football is the most popular sport with around 4 billion followers worldwide",
+        "Python language was created by Guido van Rossum.",
+    ]
+    results = evaluator.run(responses=RESPONSES)
+    ```
+    """
+
+    def __init__(
+        self,
+        instructions: str,
+        inputs: List[Tuple[str, Type]],
+        *,
+        outputs: Optional[List[str]] = None,
```

Comment:
Why not? A single key `"score"` seems like a good default to me. That default would be consistent with all the statistical metric evaluators we are adding. We agreed that all of them will have a key `"score"` in the results.

Reply:
Since the idea here is to provide a generic component that is meant to be subclassed/wrapped by another that provides specific logic to interpret/process the output of the LLM, I'd argue that this component shouldn't be opinionated on what the specific inputs and outputs should be - The details aren't part of its concern.